builtin/credential/aws/path_login_test.go

```diff
@@ -121,9 +121,110 @@ func TestBackend_pathLogin_parseIamArn(t *testing.T) {
 	}
 }
 
-func TestBackend_validateVaultHeaderValue(t *testing.T) {
+func TestBackend_validateVaultGetRequestValues(t *testing.T) {
 	const canaryHeaderValue = "Vault-Server"
-	requestURL, err := url.Parse("https://sts.amazonaws.com/")
+
+	getHeadersMissing := http.Header{
+		"Host": []string{"Foo"},
+	}
+	getHeadersInvalid := http.Header{
+		"Host":            []string{"Foo"},
+		iamServerIdHeader: []string{"InvalidValue"},
+	}
+	getHeadersValid := http.Header{
+		"Host":            []string{"Foo"},
+		iamServerIdHeader: []string{canaryHeaderValue},
+	}
+	getQueryValid := url.Values(map[string][]string{
+		"X-Amz-Algorithm":     {"AWS4-HMAC-SHA256"},
+		"X-Amz-Credential":    {"AKIDEXAMPLE/20150830/us-east-1/iam/aws4_request"},
+		"X-Amz-SignedHeaders": {"host;x-vault-aws-iam-server-id"},
+		"X-Amz-Signature":     {"5d672d79c15b13162d9279b0855cfba6789a8edb4c82c400e06b5924a6f2b5d7"},
+		"X-Amz-User-Agent":    {"aws-sdk-go-v2/1.2.0 os/linux lang/go/1.16 md/GOOS/linux md/GOARCH/amd64"},
+		"Action":              {"GetCallerIdentity"},
+		"Version":             {"2011-06-15"},
+	})
+	getQueryUnsigned := url.Values(map[string][]string{
+		"X-Amz-Algorithm":     {"AWS4-HMAC-SHA256"},
+		"X-Amz-Credential":    {"AKIDEXAMPLE/20150830/us-east-1/iam/aws4_request"},
+		"X-Amz-SignedHeaders": {"host"},
+		"X-Amz-Signature":     {"5d672d79c15b13162d9279b0855cfba6789a8edb4c82c400e06b5924a6f2b5d7"},
+		"X-Amz-User-Agent":    {"aws-sdk-go-v2/1.2.0 os/linux lang/go/1.16 md/GOOS/linux md/GOARCH/amd64"},
+		"Action":              {"GetCallerIdentity"},
+		"Version":             {"2011-06-15"},
+	})
+	getQueryNoAction := url.Values(map[string][]string{
+		"X-Amz-Algorithm":     {"AWS4-HMAC-SHA256"},
+		"X-Amz-Credential":    {"AKIDEXAMPLE/20150830/us-east-1/iam/aws4_request"},
+		"X-Amz-SignedHeaders": {"host;x-vault-aws-iam-server-id"},
+		"X-Amz-Signature":     {"5d672d79c15b13162d9279b0855cfba6789a8edb4c82c400e06b5924a6f2b5d7"},
+		"X-Amz-User-Agent":    {"aws-sdk-go-v2/1.2.0 os/linux lang/go/1.16 md/GOOS/linux md/GOARCH/amd64"},
+		"Version":             {"2011-06-15"},
+	})
+	getQueryInvalidAction := url.Values(map[string][]string{
+		"X-Amz-Algorithm":     {"AWS4-HMAC-SHA256"},
+		"X-Amz-Credential":    {"AKIDEXAMPLE/20150830/us-east-1/iam/aws4_request"},
+		"X-Amz-SignedHeaders": {"host;x-vault-aws-iam-server-id"},
+		"X-Amz-Signature":     {"5d672d79c15b13162d9279b0855cfba6789a8edb4c82c400e06b5924a6f2b5d7"},
+		"X-Amz-User-Agent":    {"aws-sdk-go-v2/1.2.0 os/linux lang/go/1.16 md/GOOS/linux md/GOARCH/amd64"},
+		"Action":              {"GetSessionToken"},
+		"Version":             {"2011-06-15"},
+	})
```

Comment:
I think there is some coverage missing for multiple actions.

Reply:
Good catch, I missed that case.